rwengine/src/audio/SoundManager.cpp

```diff
@@ -42,23 +54,124 @@ bool SoundManager::initializeOpenAL() {
     return true;
 }
 
+bool SoundManager::initializeAVCodec() {
+    av_register_all();
+
+#if RW_DEBUG && RW_VERBOSE_DEBUG_MESSAGES
+        av_log_set_level(AV_LOG_WARNING);
+#else
+        av_log_set_level(AV_LOG_ERROR);
+#endif
+
+    return true;
+}
+
 void SoundManager::SoundSource::loadFromFile(const std::string& filename) {
-    fileInfo.format = 0;
-    file = sf_open(filename.c_str(), SFM_READ, &fileInfo);
+    // Allocate audio frame
+    AVFrame* frame = av_frame_alloc();
+    if (!frame) {
+        std::cerr << "Error allocating the audio frame" << std::endl;
+        return;
+    }
+
+    // Allocate formatting context
+    AVFormatContext* formatContext = nullptr;
+    if (avformat_open_input(&formatContext, filename.c_str(), nullptr, nullptr) != 0) {
+        av_frame_free(&frame);
```

Comment:
You can use unique ptr in this case to take advantage of RAII with a custom deleted.